Lib/Database/MySqlQuery.php

```diff
@@ -0,0 +1,1247 @@
+   *
+   * @since 1.0.0
+   * @access private
+   * @var array
+   */
+  private $having = array();
+
+  /**
+   * UPDATE fields and values.
+   *
+   * Associative array `key => value`, where `key` is the fields to update, and
+   * `value` is the new value.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var array
+   */
+  private $update = array();
+
+  /**
+   * INSERT fields and values.
+   *
+   * Associative array `key => value`, where `key` is the fields to insert, and
+   * `value` is the new value.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var array
+   */
+  private $insert = array();
+
+  /**
+   * DELETE modifier.
+   *
+   * DELETE modifier can be one of the following: 'LOW_PRIORITY', 'QUICK' and
+   * 'IGNORE'.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var string
+   */
+  private $delete_modifier = '';
+
+  /**
+   * Table joins.
+   *
+   * Array of arrays, each of which contains the table to join with, the field
+   * name of the first table, the operator, the field name of the second table,
+   * and the type of the JOIN.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var array
+   */
+  private $joins = array();
+
+  /**
+   * Where clauses.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var string
+   */
+  private $where = '';
+
+  /**
+   * LIMIT.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var int
+   */
+  private $limit;
+
+  /**
+   * OFFSET.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var int
+   */
+  private $offset;
+
+  /**
+   * Query result.
+   *
+   * @since 1.0.0
+   * @access private
+   * @var mixed
+   */
+  private $result;
+
+  /**
+   * Set target table.
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @param string $name Table name.
+   * @return self
+   */
+  public function table($name) {
+    // Verify that $name is a string.
+    if (!is_string($name))
+      throw InvArgExcp::typeMismatch('table name', $name, 'String or array');
+
+    $this->table = $name;
+    return $this;
+  }
+
+  /**
+   * Set SELECT fields.
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @param array|string $field Field name.
+   * @return self
+   */
+  public function select($fields) {
+    $fields = is_array($fields) ? $fields : func_get_args();
+    $this->select = array_filter($fields, 'is_string');
+    $this->type = 'select';
+
+    return $this;
+  }
+
+  /**
+   * Set UPDATE fields and values.
+   *
+   * @since 1.0.0
+   * @access public
+   *
+   * @param array $values Associative array `key => value`, where `key` is the
+   * name of the field and `value` is the new value.
+   * @return self
+   *
+   * @throws SqueezyWeb\Exceptions\InvalidArgumentException if field names in
+   * $values aren't strings.
+   * @throws \RuntimeException if values in $values aren't scalars.
+   */
+  public function update(array $values) {
+    foreach ($values as $field => &$value) {
+      if (!is_string($field))
```

Comment:
We could skip this test. Since `$field` is an array key, it's either a string or an integer. If it's an integer, you still put it as a string in the resulting query, so it is logically correct. In the worst case the query cannot find the column, but this is not a problem that needs to be addressed by us.